### DISCUSSION TITLE
SubscribeSubject with BehaviourSubject in other workflow window

### DISCUSSION THREAD

**Discussion by sahrlund** [bonsai-rx/bonsai#1113] [Q&A]
Hi Bonsai Team, 

I really hope this question is not repetitive, but I have not managed to find a similar one, neither have I really managed to understand the logic behind this not working.

Simply, I'm working with a visual detection task, were the appearance and disappearance of a visual stimuli (drifting gratings) is triggered by an Arduino input (indicating lever push, ITI, etc). The visual stimuli is in a SelectMany node, triggered by a WindowTrigger. 

The direction (degree) of the visual stimuli is random, but I would like to save the direction for each trial (found in TakeLast) in the same csv were I save all other events. However, I can't seem to get the information of the direction of the drifting gratings displayed out of this separate workflow window. I tried broadcasting it with BehaviouralSubject, but if I put a SubscribeSubject in the same window as the Csv node, the name of my BehaviouralSubject is not there. The only seem to find each other within the same workflow. 

Is there a way to broadcast information between windows? or should I figure out a way to trigger the stimuli without SelectMany, and have everything in the same window? 

Thank you for any suggestions! (and for creating bonsai in the first place, used it in so many different ways during my phd/postdoc)

Sofie 

PS. I know I'm dragging a lot of inputs into the csv file, if you have any suggestions on what the best way of doing this is (and maybe decreasing the amount of reads of the analog input), please let me know. 

<img width="1206" alt="Capture" src="https://user-images.githubusercontent.com/97992321/200010398-9b9f5b2f-4595-4f0f-bad1-fa3e299dba94.PNG">

**Comment by bruno-f-cruz** [comment]
Heya,

Can you add the workflow? thanks!

**Reply by sahrlund** [reply]
of course, my bad!
including the arduino script with the task if you would need it for any reason, the communication between the task arduino and bonsai is with a second arduino with firmata. 
thank you for having a look at this! 
[VisDetect.Stage1.zip](https://github.com/bonsai-rx/bonsai/files/9944144/VisDetect.Stage1.zip)
